charms/tensorboards-web-app/src/charm.py

```diff
@@ -62,6 +76,13 @@ def __init__(self, *args):
             self, [http_service_port], service_name=f"{self._name}"
         )
 
+        # ambient mesh
+        if self.unit.is_leader():
+            self._mesh = ServiceMeshConsumer(self)
+            self.ingress = IstioIngressRouteRequirer(self, relation_name="istio-ingress-route")
+            self._ambient_ingress_setup()
+            self.framework.observe(self.ingress.on.ready, self._on_event)
```

Comment:
suggestion (non-blocking): since `_ambient_ingress_setup` internally checks for leadership as well, before actually doing its job, shall we skip the leadership check in here, so that we have no redundancy? That way we can also move the line `self.framework.observe(self.ingress.on.ready, self._on_event)` below after `# Set up event handlers`

(Or... as an alternative, the other way around: we don't check for leadership inside `_ambient_ingress_setup`)